### ISSUE TITLE
Some of Khorinis towers are not locked

### ISSUE THREAD

**Issue by Dehela** [Try/OpenGothic#324]
I was debating if it a bug or my memory being fuzzy but a bit of info search implies a possible bug.

So, there are six towers in Khorinis. By default those are supposed to be locked, and keys for those are distributed among some of militia soldiers. But for some reason, I have at least two of those being wide open. If I start a new game and visit the city, only one of them is open (at main gate), so I guess another one gets unlocked at some point later.

**Comment by Try** [comment]
https://user-images.githubusercontent.com/122894/192343312-505122d9-44f0-41f0-8cf4-e6cc065007d0.mp4

**Comment by Dehela** [comment]
Huh, so some guards come from towers, I wouldn't expect that. Though it's still hard to say why original game doesn't suffer from that. Could it be that in original game NPCs can close doors (tbh, I don't remember), or somehow those guards were spawned in wrong place in OpenGothic, or something totally different.

**Comment by Try** [comment]
It seems the main issue is: opengothic inserts npc, at point requested by script. This is controversial to say, but in vanilla original point means very little, as its far away from player.

When npc is far from player `daily_routine` takes over, and position is gonna be determinated by it.

Related to #295

**Comment by Try** [comment]
> [f457148](https://github.com/Try/OpenGothic/commit/f457148232a679cf9fb224ac7ccd3d7b43fa1000)

OK, that was one thing, another issue: npc can open door if they walk pass it. So in theory it's possible to force one of hotel customer to open the basement-door, just by standing on their way

**Comment by Dehela** [comment]
Would it be doable to add a check for door, if one requires a key and if yes, if fitting key is present in inventory of particular NPC?

**Comment by Try** [comment]
No, longer reproducible, after updates to daily routines handling
